compiler-rt/test/lit.common.cfg.py

```diff
@@ -371,6 +375,22 @@ def get_ios_commands_dir():
         config.compiler_rt_src_root, "test", "sanitizer_common", "ios_commands"
     )
 
+# When cmake flag to disable path search is set, symbolizer is not allowed to search in $PATH, 
+# need to specify it via XXX_SYMBOLIZER_PATH
+tool_symbolizer_path_map = {
+    "asan": "ASAN_SYMBOLIZER_PATH",
+    "hwasan": "HWASAN_SYMBOLIZER_PATH",
+    "rtsan": "RTSAN_SYMBOLIZER_PATH",
+    "tsan": "TSAN_SYMBOLIZER_PATH",
+    "msan": "MSAN_SYMBOLIZER_PATH",
+    "lsan": "LSAN_SYMBOLIZER_PATH",
+    "ubsan": "UBSAN_SYMBOLIZER_PATH"
+}
+
+if config.have_disable_symbolizer_path_search:
+    symbolizer_path = os.path.join(config.llvm_tools_dir, "llvm-symbolizer")
+    for sanitizer in tool_symbolizer_path_map.values():
+        config.environment[sanitizer] = symbolizer_path 
```

Comment:
For discussion: Instead of setting the environment variable regardless of whether it is already set, it may be better to set it only in cases where it is unset.